framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2666,7 +2669,10 @@ protected ParameterizedExecutableType methodFromUse(
             adaptGetClassReturnTypeToReceiver(methodType, receiverType, tree);
         }
 
-        methodType.setReturnType(applyCaptureConversion(methodType.getReturnType()));
+        if (inferTypeArgs) {
+            methodType.setReturnType(applyCaptureConversion(methodType.getReturnType()));
```

Comment:
This was added in https://github.com/eisop/checker-framework/commit/b420748ca3299785218f657cad76fa9963b78a30
Looking at what framework/src/main/java/org/checkerframework/framework/type/TypeFromExpressionVisitor.java method `visitMethodInvocation` does now, I think this can always be removed.

I still find it ugly that the TypeFromExpressionVisitor does the capturing and asUse, which logically I would expect to happen in `methodFromUse`.
Please file an issue that we clean this up further. In this PR, try removing this call completely.